test/cql-pytest/test_materialized_view.py

```diff
@@ -1189,3 +1192,63 @@ def test_base_partition_deletion_with_low_timestamp(cql, test_keyspace):
             # Verify we get only the row with the newer timestamp
             assert [2] == [x.c for x in cql.execute(f"SELECT c FROM {table} WHERE p=0")]
             assert [2] == [x.c for x in cql.execute(f"SELECT c FROM {mv} WHERE p=0")]
+
+# Verify that creating a materialized view with an ID that is already used fails.
+def test_creating_mv_with_existing_id(cql, test_keyspace):
+    # Returns the ID of a given CF. `cf_full_name` is of form `ks_name.cf_name`.
+    def get_id_of_cf(cf_full_name: str) -> str:
+        create_stmt = cql.execute(f'DESC TABLE {cf_full_name} WITH INTERNALS').one().create_statement
+        pattern = re.compile(r'[WITH|AND] ID = (\S+)')
+        matches = pattern.search(create_stmt)
+        match = matches.group(1)
+        assert match is not None
+        return match
+
+    with new_test_table(cql, test_keyspace, 'p int PRIMARY KEY') as table:
+        # Case 1. Try to create an MV with the ID of another MV.
+        with new_materialized_view(cql, table, '*', 'p', 'p IS NOT NULL') as mv:
+            mv_id = get_id_of_cf(mv)
+            with pytest.raises(InvalidRequest, match=f'Table with ID {mv_id} already exists: {mv}'):
+                with new_materialized_view(cql, table, '*', 'p', 'p IS NOT NULL', f'WITH ID = {mv_id}'):
+                    pass
+
+        # Case 2. Try to creat an MV with the ID of a regular table.
+        cf_id = get_id_of_cf(table)
```

Comment:
Because I think you'll need to use either DESC TABLE or DESC MATERIALIZED VIEW depending if the parameter is a table or a view (at least on Cassandra it makes a difference, IIRC), I think you should probably make that word a parameter of the get_id_of_cf() function.

Reply:
You're right. Thanks for pointing it out!